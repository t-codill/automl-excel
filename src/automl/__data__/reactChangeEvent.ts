// tslint:disable: no-any
export const reactChangeEvent: React.ChangeEvent<any> = {
    target: {}
} as unknown as React.ChangeEvent<any>;
