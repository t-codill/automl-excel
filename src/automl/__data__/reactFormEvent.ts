// tslint:disable: no-any
export const reactFormEvent: React.FormEvent<any> = {
    preventDefault(): void { return; }
} as unknown as React.FormEvent<any>;
