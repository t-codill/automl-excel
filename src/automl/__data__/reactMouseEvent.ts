// tslint:disable-next-line: no-any
export const reactMouseEvent = {} as unknown as React.MouseEvent<any>;
