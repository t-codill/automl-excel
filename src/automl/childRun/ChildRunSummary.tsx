import { RunHistoryAPIsModels } from "@vienna/runhistory";
import moment from "moment";
import * as React from "react";
import { BasicTypes } from "../common/BasicTypes";
import { IDictionary } from "../common/IDictionary";
import { calculateDuration } from "../common/utils/calculateDuration";
import { getSdkVersion } from "../common/utils/getSdkVersion";
import { BaseComponent } from "../components/Base/BaseComponent";
import { PageLoadingSpinner } from "../components/Progress/PageLoadingSpinner";
import { PropertyList } from "../components/PropertyList/PropertyList";

export interface IChildRunSummaryProps {
    experimentName: string;
    run: RunHistoryAPIsModels.MicrosoftMachineLearningRunHistoryContractsRunDetailsDto | undefined;
}

export class ChildRunSummary extends BaseComponent<IChildRunSummaryProps, {}, {}> {
    protected serviceConstructors = {};
    protected getData = undefined;
    public readonly render = (): React.ReactNode => {
        const run = this.props.run;
        if (!run || !run.properties) {
            return <PageLoadingSpinner />;
        }
        const runSumData: IDictionary<BasicTypes | React.ReactNode> = {
            "Run Id": run.runId,
            Status: run.status,
            Iteration: run.properties.iteration,
            "Run Preprocessor": run.properties.run_preprocessor,
            "Run Algorithm": run.properties.run_algorithm,
            Created: moment(run.startTimeUtc)
                .format("lll"),
            Duration: calculateDuration(run.startTimeUtc, run.endTimeUtc),
            "Sdk Version": getSdkVersion(run, "")
        };
        return <PropertyList listElements={runSumData} />;
    }
}
