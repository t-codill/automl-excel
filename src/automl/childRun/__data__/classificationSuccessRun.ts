import { NotNullableProperties } from "../../common/utils/NotNullableProperties";
import { IChildRunData } from "../ChildRun";

export const classificationSuccessRun: NotNullableProperties<IChildRunData> = {
    run: {
        runId: "AutoML_001",
        parentRunId: "AutoML_000",
        status: "Completed",
        startTimeUtc: new Date("2019-02-01T08:20:00.000Z"),
        endTimeUtc: new Date("2019-02-01T08:20:30.000Z"),
        properties: {
            iteration: "9",
            run_algorithm: "Ensemble",
            run_preprocessor: "rpp1"
        }
    },
    runMetrics: {
        scalarMet1: "1",
        scalarMet2: "2",
        scalarMet3: "3",
        accuracy_table: {},
        confusion_matrix: {}
    },
    modelUri: "sampleModelUri",
    experimentName: "classificationSuccess"
};
