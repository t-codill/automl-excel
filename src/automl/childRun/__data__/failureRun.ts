import { NotNullableProperties } from "../../common/utils/NotNullableProperties";
import { IChildRunData } from "../ChildRun";

export const failureRun: NotNullableProperties<IChildRunData> = {
    run: {
        runId: "AutoML_002",
        status: "Failed",
        parentRunId: "AutoML_000",
        startTimeUtc: new Date("2019-02-01T08:20:00.000Z"),
        endTimeUtc: new Date("2019-02-01T08:20:30.000Z"),
        properties: undefined
    },
    runMetrics: {},
    experimentName: ""
};
