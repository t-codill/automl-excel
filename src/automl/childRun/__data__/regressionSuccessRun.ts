import { NotNullableProperties } from "../../common/utils/NotNullableProperties";
import { IChildRunData } from "../ChildRun";

export const regressionSuccessRun: NotNullableProperties<IChildRunData> = {
    run: {
        runId: "AutoML_003",
        parentRunId: "AutoML_000",
        status: "Completed",
        startTimeUtc: new Date("2019-02-01T08:20:00.000Z"),
        endTimeUtc: new Date("2019-02-01T08:20:30.000Z"),
        properties: {
            iteration: "0",
            run_algorithm: "GradientBoosting",
            run_preprocessor: "StandardScalerWrapper"
        }
    },
    runMetrics: {
        residuals: {},
        predicted_true: {}
    },
    modelUri: "sampleModelUri",
    experimentName: "regressionSuccess"
};
