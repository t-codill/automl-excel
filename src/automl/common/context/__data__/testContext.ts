import { PageNames } from "../../PageNames";
import { Logger } from "../../utils/logger";
import { IWorkspaceContextProps, WorkspaceError } from "../IWorkspaceProps";
import { WorkspaceFlight } from "../WorkspaceFlight";

export function getLogCustomEventSpy(): jest.SpyInstance<ReturnType<Logger["logCustomEvent"]>> {
    const spy = jest.spyOn(Logger.prototype, "logCustomEvent");
    spy.mockImplementation(jest.fn());
    return spy;
}
let pageName: PageNames = PageNames.Unknown;
export const testContext: IWorkspaceContextProps = {
    get pageName(): PageNames { return pageName; },
    get theme(): string { return "light"; },
    discoverUrls: {},
    flight: new WorkspaceFlight(""),
    location: "eastus",
    resourceGroupName: "testResource",
    subscriptionId: "00000000-0000-0000-0000-000000000000",
    logger: new Logger("test"),
    getToken: () => "testToken",
    setPageName: (value: PageNames) => { pageName = value; },
    workspaceName: "testWorkSpace",
    setNavigationBarButtons(): void { return; },
    onError: (err: WorkspaceError) => { throw err; },
    clearErrors: () => { return; },
    setHeader: () => { return; },
    setLoading: () => { return; }
};
