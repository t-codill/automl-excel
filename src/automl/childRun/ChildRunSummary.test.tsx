import { shallow } from "enzyme";
import * as React from "react";
import { classificationSuccessRun } from "./__data__/classificationSuccessRun";
import { failureRun } from "./__data__/failureRun";
import { regressionSuccessRun } from "./__data__/regressionSuccessRun";
import { ChildRunSummary } from "./ChildRunSummary";

describe("Child Run Summary", () => {
    it("should render loading image when no run metrics", async () => {
        const tree = shallow(<ChildRunSummary run={undefined} modelUri={undefined} experimentName={""} modelId={undefined} />);
        expect(tree)
            .toMatchInlineSnapshot("<pageLoadingSpinner />");
    });
    it("should render success data without download link", async () => {
        const tree = shallow(<ChildRunSummary run={classificationSuccessRun.run} modelUri={undefined} experimentName={""} modelId={undefined} />);
        expect(tree)
            .toMatchSnapshot();
    });
    it("should render success data with download link", async () => {
        const tree = shallow(<ChildRunSummary run={regressionSuccessRun.run} modelUri={regressionSuccessRun.modelUri} experimentName={"Experiment"} modelId={undefined} />);
        expect(tree)
            .toMatchSnapshot();
    });
    it("should render page loading image without run properties", async () => {
        const tree = shallow(<ChildRunSummary run={failureRun.run} modelUri={failureRun.modelUri} experimentName={"Experiment"} modelId={undefined} />);
        expect(tree)
            .toMatchInlineSnapshot("<pageLoadingSpinner />");
    });
});
