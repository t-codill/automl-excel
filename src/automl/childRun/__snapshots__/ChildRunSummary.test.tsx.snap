// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Child Run Summary should render success data with download link 1`] = `
<PropertyList
  listElements={
    Object {
      "Created": "Feb 1, 2019 7:20 PM",
      "Duration": "00:00:30",
      "Iteration": "0",
      "Run Algorithm": "GradientBoosting",
      "Run Id": "AutoML_003",
      "Run Preprocessor": "StandardScalerWrapper",
      "Status": "Completed",
    }
  }
/>
`;

exports[`Child Run Summary should render success data without download link 1`] = `
<PropertyList
  listElements={
    Object {
      "Created": "Feb 1, 2019 7:20 PM",
      "Duration": "00:00:30",
      "Iteration": "9",
      "Run Algorithm": "Ensemble",
      "Run Id": "AutoML_001",
      "Run Preprocessor": "rpp1",
      "Status": "Completed",
    }
  }
/>
`;
